javaparser-testing/src/test/java/com/github/javaparser/ast/type/TypeTest.java

```diff
@@ -53,7 +53,7 @@ public void arrayType() {
         assertTrue(type.isArrayType());
         ArrayType arrayType = type.asArrayType();
         final ArrayType[] s = new ArrayType[1];
-        type.ifArrayType(t -> s[0] = t);
+        type.ifArrayType(t -> s[0] = (ArrayType)t);
```

Comment:
Huh? The point is that a cast isn't necessary

Reply:
Yes, but I could not convince the compiler that was the case... it kept telling me it was an Object, so I just made it happy to move forward this thing and get initial feedback, just in case we figured out immediately I was going in the wrong direction

Reply:
Hmmmm, that needs an issue, although I don't know how to reproduce it.

Reply:
If you remove it and it does compile it for you then it has probably something to do with my checkout (even if I have no idea what it could be)

Reply:
Yeah, I don't check in broken code ;-) Also, Travis is completely fine with it, and since it is a test I would prefer if you don't change it :-)

Reply:
I removed the hack and it is failing also on the server. Really strange...

Reply:
This works:
```
arrayType.ifArrayType(t -> s[0] = t);
```

but this does not:
```
type.ifArrayType(t -> s[0] = t);
```

However both versions of ifArrayType have the same signature...

Reply:
This also works:
```
        Consumer<ArrayType> c = t -> s[0] = t;
        type.ifArrayType(c);
```

Reply:
Oh my...

This works:
```
    @Test
    public void arrayType() {
        Type<?> type = parseType("int[]");
        assertTrue(type.isArrayType());
        ArrayType arrayType = type.asArrayType();
        final ArrayType[] s = new ArrayType[1];
        type.ifArrayType(t -> s[0] = t);
        assertNotNull(s[0]);
    }
```

This does not:
```
    @Test
    public void arrayType() {
        Type type = parseType("int[]");
        assertTrue(type.isArrayType());
        ArrayType arrayType = type.asArrayType();
        final ArrayType[] s = new ArrayType[1];
        type.ifArrayType(t -> s[0] = t);
        assertNotNull(s[0]);
    }
```

`Type` != `Type<?>`...